### DISCUSSION TITLE
JsonRpcConnectionHandler deprecation: how to migrate?

### DISCUSSION THREAD

**Discussion by dmm9** [eclipse-theia/theia#13417] [Q&A]
Changing `JsonRpcConnectionHandler` to `RpcConnectionHandler` because of deprecation notice, but then RPC stops working.. I cannot find any documentation about how to migrate the interface. What else should be changed, apart from the name??????

Working code:
```typescript
bind(ConnectionHandler).toDynamicValue( ctx =>
        new JsonRpcConnectionHandler(workspaceControllerPath, () =>
            ctx.container.get(WorkspaceControllerApi)
        )
    ).inSingletonScope();
```
After renaming JsonRpc* to Rpc*... I get the runtime error when the RPC is called:
```shell
ERROR Request loadFile failed with error: no target was set to handle loadFile Error: no target was set to handle loadFile
    at RpcProxyFactory.onRequest ...
```

**Comment by msujew** [comment]
Hey @dmm9,

not sure what to tell you, we are using it ourselves successfully:

https://github.com/eclipse-theia/theia/blob/df6341a4aecec7846cf72cf9f1fdd1c2b5bc5e36/packages/core/src/node/backend-application-module.ts#L90-L94

Perhaps @tortmayr knows more?
